components/link-button.js

```diff
@@ -0,0 +1,156 @@
+const template = document.createElement("template");
+template.innerHTML = `
+<link rel="stylesheet" href="./global-styles.css" />
+<style>
+  /* Default: Extra-small devices such as small phones (less than 640px) */
+  a {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: max-content;
+    text-align: center;
+  }
+
+  a.big.ghost {
+    border: 1px solid #000;
+  }
+
+  a.ghost {
+    background-color: transparent;
+    border: none;
+    color: #000;
+  }
+
+  a.secondary {
+    background-color: #d9d9d9;
+    color: #000;
+  }
+
+  a.small {
+    font-size: 13px;
+    padding: 10px 20px;
+    height: 35px;
+    border-radius: 7px;
+  }
+
+  a.big {
+    padding: 10px 25px;
+    font-size: 14px;
+    height: 45px;
+    border-radius: 10px;
+  }
+
+  /* Small devices such as large phones (640px and up) */
+  @media only screen and (min-width: 640px) {
+    /* TODO: fill after the design update */
+  }
+
+  /* Medium devices such as tablets (768px and up) */
+  @media only screen and (min-width: 768px) {
+    a.small {
+      padding: 5px 25px;
+      font-size: 14px;
+    }
+
+    a.big {
+      padding: 16px 40px;
+      font-size: 20px;
+    }
+  }
+
+  /* Large devices such as laptops (1024px and up) */
+  @media only screen and (min-width: 1024px) {
+    a.small {
+      padding: 5px 30px;
+      font-size: 16px;
+    }
+
+    a.big {
+      padding: 16px 40px;
+      font-size: 20px;
+    }
+  }
+
+  /* Largest devices such as desktops (1280px and up) */
+  @media only screen and (min-width: 1280px) {
+    /* TODO: fill after the design update */
+  }
+</style>
+<a>
+  <slot></slot>
+</a>
+`;
+
+class LinkButton extends HTMLElement {
+  constructor() {
+    super();
+
+    this.attachShadow({ mode: "open" });
+    this.shadowRoot.appendChild(template.content.cloneNode(true));
+
+    this.setDefaultAttributes();
+    this.validateAttributes();
+    this.updateAttributes();
+  }
+
+  validateAttributes() {
+    if (!this.hasAttribute("href")) {
+      throw new Error('The "href" attribute is required.');
+    }
+
+    if (this.hasAttribute("size")) {
+      const size = this.getAttribute("size");
+      const validAttributes = ["big", "small"];
+
+      if (size !== "big" && size !== "small") {
```

Comment:
nit: 요게 살짝 덜 redundant할 것 같아요.

```suggestion
      if (!validAttributes.includes(size)) {
```

Reply:
memo: nit는 nitpick의 약자 ... 아주 작은 개선 사항을 제한할 때 사용! (맞을까요?)

Reply:
@DaleSeo 반영했습니다~